operator/v1/stable.ingresscontroller.testsuite.yaml

```diff
@@ -13,3 +13,84 @@ tests:
       kind: IngressController
       spec:
         httpEmptyRequestsPolicy: Respond
+  - name: Should not allow to set/delete HSTS header.
+    initial: |
+      apiVersion: operator.openshift.io/v1
+      kind: IngressController
+      metadata:
+        name: default-hsts
+        namespace: openshift-ingress-operator
+      spec:
+        httpHeaders:
+          actions:
+            response:
+            - name: X-Frame-Options
+              action:
+                type: Set
+                set:
+                  value: DENY
+            - name: Strict-Transport-Security
+              action:
+                type: Delete
+            request:
+            - name: Content-Location
+              action:
+                type: Set
+                set:
+                  value: /my-first-blog-post
+            - name:   Content-Language
+              action:
+                type: Delete
+    expectedError: "strict-transport-security may not be set via header actions"
+  - name: Should not allow to set/delete dynamic headers with unclosed braces.
+    initial: |
+      apiVersion: operator.openshift.io/v1
+      kind: IngressController
+      metadata:
+        name: default-unclosed-braces
+        namespace: openshift-ingress-operator
+      spec:
+        httpHeaders:
+          actions:
+            response:
+            - name: expires
+              action:
+                type: Set
+                set:
+                  value: "%[date(3600),http_date"
+            request:
+            - name: Content-Location
+              action:
+                type: Set
+                set:
+                  value: /my-first-blog-post
+            - name:   Content-Language
+              action:
+                type: Delete
+    expectedError: "in body should match '^(%(%|(\\{[-+]?[QXE](,[-+]?[QXE])*\\})?([A-Za-z]+|\\[[.0-9A-Z_a-z]+(\\([^)]+\\))?(,[.0-9A-Z_a-z]+(\\([^)]+\\))?)*\\]))|[^%[:cntrl:]])*$'"
+  - name: Should not allow certain converters for dynamic values.
+    initial: |
+      apiVersion: operator.openshift.io/v1
+      kind: IngressController
+      metadata:
+        name: default-allowed-converters
+        namespace: openshift-ingress-operator
+      spec:
+        httpHeaders:
+          actions:
+            response:
+            - name: map
+              action:
+                type: Set
+                set:
+                  value: "%[path,map_beg(/etc/hapee-2.6/routes.map,webservers)]"
+            request:
+            - name: Content-Location
+              action:
+                type: Set
+                set:
+                  value: /my-first-blog-post
+            - name:   Content-Language
+              action:
+                type: Delete
+    expectedError: "converter specified not permitted."
```

Comment:
How many converters are permitted? 

Also, please restore the new line at the end of this file, it will create a weird git diff in the future otherwise

Reply:
Hi @JoelSpeed ! I am checking with @Miciah on whether we shall have a allowed converter list or whether we shall just have a format to check this the value. @Miciah Do we need an allowed converter list ? I think even if we just check the format of the value it shall be fine as some converters may deprecate in the future and also there hundreds if converters available so it is very hard to choose to pick from them.

Reply:
Just to tie up loose ends, we discussed it and decided that we would have an explicit list of allowed converters, and an explicit list of allowed sample fetchers.  The question and decision are documented by the EP under "Open Questions", sub-heading "Should this enhancement allow setting dynamic header values?".